### PR TITLE
Replaced floating button with tabs.

### LINKED ISSUE #16: Replace floating button with tabs

### PR DESCRIPTION
**Pull request by genaforvena**
Fixes #16

### CODE REVIEW

**Comment by genaforvena** [comment]
@sanovikov71

File: app/build.gradle [hunk 1]
```diff
@@ -78,7 +78,9 @@ dependencies {
     compile 'io.reactivex:rxjava:1.1.0'
     compile 'io.realm:realm-android:0.87.4'
 
-    compile 'com.android.support:recyclerview-v7:23.0.1'
+    compile 'com.android.support:recyclerview-v7:23.1.1'
```
Comment (sanovikov71) [inline app/build.gradle:86]: Two recyclerview dependencies in build.gradle

Comment (genaforvena) [inline app/build.gradle:86]: Incremented version.
Old one was deleted.

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                    listOfArtObjects
+                }
+    }
+
+    fun getArtObject(id: String): Observable<ArtObjectUi> {
+        return realm
+                .where(RealmArtObject::class.java)
+                .equalTo("id", id)
+                .findFirst()
+                .asObservable<RealmObject>()
```
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt:38]: Why is it generic here? And why the Ttype is RealmObject?

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        val realmAuthor = RealmAuthor()
+        realmAuthor.id = "1"
+        realmAuthor.name = "Vasya"
+        realmAuthor.description = "Description"
+
+        val realmArtObject = RealmArtObject()
+        realmArtObject.author = realmAuthor
+        realmArtObject.description = "Description"
+        realmArtObject.name = "Name"
+        realmArtObject.id = SystemClock.currentThreadTimeMillis().toString()
```
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt:53]: Maybe we should change id generation to simplify testing?

File: app/src/main/java/org/imozerov/streetartview/ui/ExploreArtActivity.kt [hunk 4]
```diff
@@ -3,47 +3,88 @@ package org.imozerov.streetartview.ui
...
     }
 
     override fun openArtObjectDetails(id: String?) {
+        Log.d(TAG, "openArtObjectDetails($id)")
         val intent = Intent(this, DetailArtObjectActivity::class.java)
         intent.putExtra(DetailArtObjectActivity.EXTRA_KEY_ART_OBJECT_DETAIL_ID, id)
         startActivity(intent)
     }
 
+    internal class Adapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
```
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/ExploreArtActivity.kt:68]: Why do not put it in a separate file?

Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/ExploreArtActivity.kt:68]: And change the name to smth meaningful

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import org.imozerov.streetartview.StreetArtViewApp
+import org.imozerov.streetartview.storage.DataSource
+import org.imozerov.streetartview.ui.interfaces.Filterable
+import org.imozerov.streetartview.ui.model.ArtObjectUi
+import rx.Subscription
+import rx.android.schedulers.AndroidSchedulers
+import javax.inject.Inject
+
+class ArtListFragment : Fragment(), Filterable {
+    private val TAG = "AlertListFragment"
```
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt]: Typo change AlertListFragment to ArtListFragment. Perhaps, it is better to use smth like ArtListFragment.class.getSimpleName() kotlin equivalent?

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        super.onActivityCreated(savedInstanceState)
+        (activity.application as StreetArtViewApp).storageComponent.inject(this)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val rootView = inflater!!.inflate(R.layout.fragment_art_list, container, false)
+        val context = context
+
+        adapter = ArtListAdapter(getContext(), java.util.ArrayList<ArtObjectUi>())
```
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt:33]: java.util is ugly

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        fetchSubscription = startFetchingArtObjectsFromDataSource()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        fetchSubscription?.unsubscribe()
+    }
+
+    override fun applyFilter(queryToApply: String) {
+        listFilterQuery = queryToApply
```
Comment (sanovikov71) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt]: Can it be done without a field? To pass query directly to startFetching...

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt (hunk 2 above)
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt:38]: Compiler failed to detect type of findFirst return object. We need to help him do it.

File: app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt (hunk 3 above)
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/storage/DataSource.kt:53]: oh common. we'll remove it in 2-3 commits. Let it be as it is

File: app/src/main/java/org/imozerov/streetartview/ui/ExploreArtActivity.kt (hunk 4 above)
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/ExploreArtActivity.kt:68]: When the file will grow we'll do so. Not now. For now the name is meaningful enough as it is internal class.

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt (hunk 5 above)
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt]: logt in android studio generates tag with string literal. I prefer to stick to this strategy. I'll fix the typo.

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt (hunk 6 above)
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt:33]: I wanted to explicitly show java classes that are used.
Also android studio imports java.util.\* so it prevents us from importing Observable from rx. java.util.Observable is suggested instead.

File: app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt (hunk 7 above)
Comment (genaforvena) [inline app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListFragment.kt]: Will be fixed in the next commit. But it will still be the field.

**Pull request merged**